src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py

```diff
@@ -19,11 +19,23 @@
 
 LOGGER = logging.getLogger(__name__)
 
+
+# Handle optional import. The azure libraries are only present if
+# promptflow-azure is installed.
 try:
-    from azure.storage.blob import BlobServiceClient
     from azure.ai.ml.entities._credentials import AccountKeyConfiguration
-except ImportError:
-    LOGGER.debug("promptflow.azure is not installed.")
+    from azure.ai.ml.entities._datastore.datastore import Datastore
+    from azure.storage.blob import BlobServiceClient
+except (ModuleNotFoundError, ImportError):
+    # If the above mentioned modules cannot be imported, we are running
+    # in local mode and MLClient in the constructor will be None, so
+    # we will not arrive to Azure-dependent code.
+    LOGGER.warning(

```

Comment:
Hope user won't see this warning when `azure_ai_project` is not given. Please check.

Reply:
In this cease we should not log this error, as if user has not provided `azure_ai_project`, the azure-related code will not be triggered and user will get the warning about tracking_uri is not being provided. If, on the other hand, use will provide project information, it will cause exception during [azure]PFClient initialization.